pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -99,12 +99,16 @@ func (admitter *VMsAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
-	causes := admitter.applyInstancetypeToVm(&vm)
+	// We apply any referenced instancetype and preferences early here to the VirtualMachine in order to
+	// validate the resulting VirtualMachineInstanceSpec below. As we don't want to persist these changes
+	// we pass a copy of the original VirtualMachine here and to the validation call below.
+	vmCopy := vm.DeepCopy()
+	causes := admitter.applyInstancetypeToVm(vmCopy)
```

Comment:
I think its better to move the 
vmCopy := vm.DeepCopy()
causes := admitter.applyInstancetypeToVm(vmCopy)
 to inside ValidateVirtualMachineSpec before ValidateVirtualMachineInstanceSpec

Reply:
That would mean refactoring `ValidateVirtualMachineSpec` quite heavily and adjusting callers such as `ValidateVMPoolSpec` as `applyInstancetypeToVm` needs a reference to the overall `VirtualMachine` in order to work, I don't think it's worth it IMHO.